src/http/validations/device.js

```diff
@@ -13,6 +14,19 @@ const getSingleDevice = (req, res, next) => {
   return next();
 };
 
-export default {
-  getSingleDevice,
+const updateDevice = (req, res, next) => {
+  const { deviceId } = req.params;
+
+  const fieldErrors = new FieldErrors();
+
+  if (!db.validResourceId(deviceId)) fieldErrors.addError('deviceId', 'deviceId should be a ');
+  if (!deviceId || typeof (deviceId) !== 'string') fieldErrors.addError('deviceId', 'deviceId is a required string');
+  if (req.body.constructor === Object && Object.keys(req.body).length === 0) fieldErrors.addError('request', 'Request body can\'t be empty');
```

Comment:
The `fieldErrors` is such that if used like this `fieldErrors(a, b)`; a should be a field in the request body while b should be the error associated with that field. `reqeust` is not a field in the request body so please revise this validation.

Reply:
fixed.
```
if (label) {
    if (typeof (label) !== 'string') fieldErrors.addError('label', 'label is a string');
  }
  if (wardId) {
    if (!db.validResourceId(wardId)) fieldErrors.addError('wardId', 'wardId is not valid');
  }
  if (hospitalId) {
    if (!db.validResourceId(hospitalId)) fieldErrors.addError('hospitalId', 'hospitalId is not valid');
  }
  if (!deviceId) fieldErrors.addError('deviceId', 'deviceId is required');
  if (!db.validResourceId(deviceId)) fieldErrors.addError('deviceId', 'deviceId is not valid');
  
  if (req.body.constructor === Object && Object.keys(req.body).length === 0) {
    return res.status(400).json({ success: false, message: 'Invalid request: All fields can\'t be empty' });
  }
```